.clang-format

```diff
@@ -67,7 +67,7 @@ IncludeCategories:
   - Regex:           '.*'
     Priority:        1
 IndentCaseLabels: false
-IndentWidth:     8
+IndentWidth:     4
```

Comment:
8 to 4 space indentation is the major change that I observed as the major change in the diff, is it possible to keep it as 8 for one commit and later change only the indent to 4 and run the clang and provide those changes as another commit?

Reply:
I'm with Hari - I actually wonder if we should make two separate changes here. It's going to be pretty much impossible to notice a functional change in this enormous patch, but even harder if it touches literally every line because of the indenting change.

Reply:
Ok. I've rewritten history here so there's a commit where I've applied the original Clang Format style and then another commit where I've applied our agreed changes (mostly just indentation).

Before applying agreed changes: d66812e
After applying agreed changes: f3fa720

@agorrod @kommiharibabu Is this just for the purposes of this pull request? Or do you want the actual commits that go into `develop` to look like this? To me, it just creates yet another inconvenient commit that gets in the way when blaming a line.

Reply:
@tetsuo-cpp I don't have any problem in squashing the commits before merging.
With seperate commit in pull request will help the reviewers to find out exactly what is changed by the clang other than indentation.

Reply:
@tetsuo-cpp @agorrod I just want to hear your opinions on the backport tickets after the changes of tabs from 8 to 4, this may introduce some additional conflicts?

Reply:
Sorry for taking a long time to respond to this.
The way the server teams did this was to apply the formatter to each of the release branches. This way, you will have a window in time where merge conflicts can happen (backporting a pre-format `develop` change to `mongodb-4.0`). But any new post-format changes going into develop, which will be the majority of backports going forwards, shouldn't be affected.

Reply:
@tetsuo-cpp I am happy that if we take care of applying the clang format to all the supported branches. If we can merge the clang format changes when there are no planned minor releases,  we can delay the backport fixes until the clang format changes are done in supported branches also.